boards/arc/iotdk/arc_mpu_regions.c

```diff
@@ -19,6 +19,20 @@ static struct arc_mpu_region mpu_regions[] = {
 			 DT_REG_ADDR(DT_INST(0, arc_dccm)),
 			 DT_REG_SIZE(DT_INST(0, arc_dccm)),
 			 REGION_KERNEL_RAM_ATTR),
+	/* Region XCCM */
+#if DT_REG_SIZE(DT_INST(0, arc_xccm)) > 0
+	MPU_REGION_ENTRY("XCCM",
+			 DT_REG_ADDR(DT_INST(0, arc_xccm)),
+			 DT_REG_SIZE(DT_INST(0, arc_xccm)),
+			 REGION_KERNEL_RAM_ATTR | REGION_DYNAMIC),
```

Comment:
iotdk has an MPU that supports overlapping regions, I don't think REGION_DYNAMIC is required (DCCM doesn't have the attribute either).
For the other boards, I can see that access may be restricted to certain tasks only, so dynamic allocation and switching access rights in context switches would be required.